tscat/base.py

```diff
@@ -193,10 +196,16 @@ def __setattr__(self, key, value):
             if value < self.start:
                 raise ValueError("stop date has to be after start date")
         elif key in ['tags', 'products']:
-            if any(type(v) != str for v in value):
+            if any(not isinstance(v, str) for v in value):
```

Comment:
Looking at this, I wonder if using properties isn't better, each property would holds its own logic.

Reply:
One requirement for tscat was to be able to do `event.aribtrary_field = "value"`  for dynamic attributes and fixed-attributes. At the time of writing the `__setattr__`-function it was the only way I found.

I might have to look at the `@property`-decorator again, but AFAICR the only way to set a property is by using a setter-function? Or does this allow also the use of the  assignment operator.